infinyon/fluvio
# [Merged by Bors] - Introduce channels into CLI

This adds the concept of separate `channels` in the CLI, with the ability to change between them.

3 default channels:
* `stable`
* `latest`
* `dev`

By default, we use the `stable` channel, but you can switch to another channel with:

`fluvio version switch <channel>`

To add a version channel, you need to create it first

For example, to create a channel for `0.9.15` run
`fluvio version create 0.9.15`

Then you can switch over to it with

`fluvio version switch 0.9.15`

---

Simplifies starting a cluster to `fluvio cluster start` for all channels (only if using the fluvio-channel binary)

Changes `fluvio-cli` to respect 3 environment variables to control cluster and plugin interactions
* `FLUVIO_RELEASE_CHANNEL`
* `FLUVIO_EXTENSIONS_DIR`
* `FLUVIO_IMAGE_TAG_STRATEGY`

Manual tests:
- Copy `fluvio-channel` binary `~/.fluvio/bin/fluvio` 
- Switch to stable channel `fluvio version switch stable`, start a cluster with `fluvio cluster start`
- Switch to latest, `fluvio version switch latest`, start a cluster with `fluvio cluster start`
- Verify the image that is used for pods is correct

Additionally, modifies `install.sh` to support the installation of `fluvio-channel` to `~/.fluvio/bin/fluvio`, and a target fluvio binary as `~/.fluvio/bin/fluvio-<channel>`

Resolves #1663
